### PR TITLE
846 make /upload to match toFHIR logic 

### PR DESCRIPTION
**Pull request by leite08**
Ref: https://github.com/metriport/metriport/issues/846

### Dependencies

none

### Description

- make /upload to match toFHIR logic 
- doc ref status now matches what we get from CW - [link](https://github.com/metriport/metriport/pull/889/files#diff-ca68a1648769cbb4448dba2cab721fd02d62e511904af21aba6e0e7829aa9ee0R117)
- improve error handling

### Release Plan

- nothing special, asap

### CODE REVIEW

**Review by RamilGaripov — APPROVED** [review]
LGTM. One nit: add the ticket ref in the PR description

File: packages/api/src/command/medical/admin/upload-doc.ts [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 import { metriportDataSourceExtension } from "../../../external/fhir/shared/extensions/metriport";
 import { Config } from "../../../shared/config";
-import { getOrganizationOrFail } from "../organization/get-organization";
+import { randomInt } from "../../../shared/numbers";
 import { getPatientOrFail } from "../patient/get-patient";
 
 const apiUrl = Config.getApiUrl();
 const docContributionUrl = `${apiUrl}/doc-contribution/commonwell/`;
 
+const smallId = () => String(randomInt(3)).padStart(3, "0");
```
Comment (Goncharo) [inline packages/api/src/command/medical/admin/upload-doc.ts:13]: nit: `3` could go in a named const

File: packages/api/src/command/medical/admin/upload-doc.ts [hunk 2]
```diff
@@ -32,34 +34,42 @@ export async function createAndUploadDocReference({
...
-  const organization = await getOrganizationOrFail({ cxId });
 
   const fhirApi = makeFhirApi(cxId);
-
-  const now = dayjs();
+  const refDate = dayjs();
+  const orgId = smallId();
+  const orgRef = `org${orgId}`;
+  const practitionerId = smallId();
+  const practitionerRef = `org${practitionerId}`;
```
Comment (Goncharo) [inline packages/api/src/command/medical/admin/upload-doc.ts:43]: should this be `practitioner${practitionerId}` ?

File: packages/api/src/command/medical/admin/upload-doc.ts [hunk 3]
```diff
@@ -32,34 +34,42 @@ export async function createAndUploadDocReference({
...
       },
       {
-        resourceType: "Patient",
-        id: patient.id,
+        resourceType: "Practitioner",
+        id: practitionerRef,
+        name: [
+          {
+            family: `Last ${practitionerId}`,
+            given: [`First ${practitionerId}`],
```
Comment (Goncharo) [inline packages/api/src/command/medical/admin/upload-doc.ts:70]: don't understand why we have Last/First in the name - isn't this implicit based on the family/given property?

Comment (Goncharo) [reply packages/api/src/command/medical/admin/upload-doc.ts:70]: can ignore above, just realized this is a func for testing

**Review by Goncharo — APPROVED** [review]
LGTM assuming this is resolved if needed https://github.com/metriport/metriport/pull/889/files#r1323752523

File: packages/api/src/command/medical/admin/upload-doc.ts (hunk 2 above)
Comment (leite08) [reply packages/api/src/command/medical/admin/upload-doc.ts:43]: good catch

I had this fixed on this downstream PR: https://github.com/metriport/metriport/pull/890/files#diff-3d42b174b9456e6cba24c22730804f621f249c1013c2aedac4adc1c83c3635daR45

File: packages/api/src/command/medical/admin/upload-doc.ts (hunk 1 above)
Comment (leite08) [reply packages/api/src/command/medical/admin/upload-doc.ts:13]: yeah, but its just an internal endpoint and this is only used here, so for the sake of finishing this before getting AFK I'll leave like this :)

**Pull request merged**